events-api/src/modules/events/events.service.ts

```diff
@@ -48,21 +49,82 @@ export class EventsService {
   }
 
   async listEvents(searchParams: ListEventsParamsReqDto) {
-    const page = searchParams.page ?? 1;
-    const limit = searchParams.limit ?? 20;
+    const { page = 1, limit = 20, sort_by, sort_order = 'asc', name  } = searchParams
 
     const events = await this.databaseService.event.findMany({
       where: {
-        eventName: searchParams.name
-          ? {
-              contains: searchParams.name,
-            }
+        eventName: name
+          ? { contains: name }
           : undefined,
       },
+
+      orderBy: sort_by && sort_order
+        ? { [sort_by]: sort_order }
+        : undefined,
+
       skip: (page - 1) * limit,
       take: limit,
     });
 
     return events;
   }
+
+
+  async updateEvent(eventId: string, dto: UpdateEventReqDto) {
+    const { eventDate, eventDescription, eventLocation, eventName } = dto
+
+    const existingEventById = await this.databaseService.event.findFirst({
+      where: {
+        eventId,
+      },
+    });
+
+    if (!existingEventById) {
+      throw new BadRequestException('Event not found');
+    }
+
+    const dataToUpdate: any = {};
+
+    if (eventDate) {
```

Comment:
Prisma can handle undefined values pretty good

Reply:
Removed this unnecessary checks